packages/roosterjs-editor-plugins/lib/plugins/ContentEdit/features/textFeatures.ts

```diff
@@ -56,6 +56,32 @@ const IndentWhenTabText: BuildInEditFeature<PluginKeyboardEvent> = {
     },
 };
 
+/**
+ * Requires @see ExperimentalFeatures.TabKeyTextFeatures to be enabled
+ * If Whole Paragraph selected, outdent paragraph on Tab press
+ */
+const OutdentWhenTabText: BuildInEditFeature<PluginKeyboardEvent> = {
+    keys: [Keys.TAB],
+    shouldHandleEvent: (event, editor) => {
+        const selection = editor.getSelectionRangeEx();
+
+        return (
+            editor.isFeatureEnabled(ExperimentalFeatures.TabKeyTextFeatures) &&
+            selection.type == SelectionRangeTypes.Normal &&
+            !selection.areAllCollapsed &&
+            event.rawEvent.shiftKey &&
```

Comment:
Check this first, and even before we call getSelectionRangeEx since it is the fastest check and in most cases it will return false